src/editors/sharedComponents/TinyMceWidget/hooks.js

```diff
@@ -116,7 +102,22 @@ export const replaceStaticWithAsset = ({
 
       // assets in expandable text areas do not support relative urls so all assets must have the lms
       // endpoint prepended to the relative url
-      if (editorType === 'expandable') {
+      if (isLibraryKey(learningContextId)) {
+        // This function doesn't currently know how to deal with Library assets.
+        // Libraries don't mangle the path into an asset key–it might be sufficient
+        // to remove the initial "/" in a "/static/images/foo.png" link, and then
+        // set the base URL to the correct ComponentVersion base. If we let this
+        // function try to deal with Library assets, it would convert them in such a
+        // way that it wouldn't convert back later on, and we'd end up storing the
+        // incorrect OLX and breaking the display from that point forward.
+        //
+        // So until we handle it better, just disable static asset URL substitutions
+        // when dealing with Library content.
```

Comment:
This is no longer accurate, right? Now that your code is dealing with the case properly?

Reply:
Yes, I'll update this comment with the details of the solution